features/checkout/prevent_skipping_checkout_steps.feature

```diff
@@ -33,3 +36,29 @@ Feature: Prevent skipping checkout steps
         And I am at the checkout addressing step
         When I want to complete checkout
         Then I should be on the checkout addressing step
+
+    @ui
+    Scenario: Skipping addressing checkout step when order total is zero
+        Given I have product "PHP T-Shirt" in the cart
+        And I have product "Paganini T-Shirt" in the cart
+        And I am at the checkout addressing step
+        When I want to complete checkout
+        Then I should be on the checkout addressing step
+
+    @ui
+    Scenario: Skipping shipping checkout step when order total is zero
+        Given I have product "PHP T-Shirt" in the cart
+        And I have product "Paganini T-Shirt" in the cart
+        And I specified the shipping address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        When I want to complete checkout
+        Then I should be on the checkout shipping step
+
+    @ui
+    Scenario: Not being able go to payment checkout step when order total is zero
```

Comment:
Maybe, but from the point of view of a customer in this scenario the payment method exist (the customer think that it exist), so they are not able to go to this step.